CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
 
 * Font-lock Lein's `defproject` as a keyword.
 
+### Bugs fixed
+
+* [#586](https://github.com/clojure-emacs/clojure-mode/issues/586): Fix infinite loop when opening file containing `comment` with `clojure-toplevel-inside-comment-form` set to `t`.
```

Comment:
Btw, is this specific to `comment` or to just having a single form in the buffer? Looking at the code it's the latter.